File: app/src/main/java/com/tuvakov/zeyoube/android/utils/PrefUtils.java
package com.tuvakov.zeyoube.android.utils;

import android.content.SharedPreferences;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class PrefUtils {

    private static final String PREF_ACCOUNT_NAME = "accountName";
    private static final String PREF_LAST_SYNC_TIME = "lastSyncTime";

    private final SharedPreferences mSharedPreferences;

    @Inject
    public PrefUtils(SharedPreferences sharedPreferences) {
        mSharedPreferences = sharedPreferences;
    }


    public String getAccountName() {
        return mSharedPreferences.getString(PREF_ACCOUNT_NAME, null);
    }

    public void saveAccountName(String accountName) {
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        editor.putString(PREF_ACCOUNT_NAME, accountName);
        editor.apply();
    }

    public void deleteAccountName() {
        saveAccountName(null);
    }

    public long getLastSyncTime() {
        return mSharedPreferences.getLong(PREF_LAST_SYNC_TIME, 0);
    }

    public void saveLastSyncTime(long lastSyncTime) {
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        editor.putLong(PREF_LAST_SYNC_TIME, lastSyncTime);
        editor.apply();
    }
}

File: app/src/main/java/com/tuvakov/zetube/android/utils/PrefUtils.kt
package com.tuvakov.zetube.android.utils

import android.content.SharedPreferences
import androidx.core.content.edit
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PrefUtils @Inject constructor(private val mSharedPreferences: SharedPreferences) {
    val accountName: String?
        get() = mSharedPreferences.getString(PREF_ACCOUNT_NAME, null)

    fun saveAccountName(accountName: String?) {
        mSharedPreferences.edit { putString(PREF_ACCOUNT_NAME, accountName) }
    }

    fun deleteAccountName() {
        saveAccountName(null)
    }

    val lastSyncTime: Long
        get() = mSharedPreferences.getLong(PREF_LAST_SYNC_TIME, 0)

    fun saveLastSyncTime(lastSyncTime: Long) {
        mSharedPreferences.edit { putLong(PREF_LAST_SYNC_TIME, lastSyncTime) }
    }

    companion object {
        private const val PREF_ACCOUNT_NAME = "accountName"
        private const val PREF_LAST_SYNC_TIME = "lastSyncTime"
    }

}